UmpleToJava/UmpleTLTemplates/JavaClassGenerator.ump

```diff
@@ -98,17 +98,27 @@ public <<# if (uClass.getIsAbstract()) { append(realSb, "{0} ", "abstract"); } #
 for (StateMachine smq : uClass.getStateMachines())
   {
     if (smq.isQueued())
-    {
-      append(realSb," implements Runnable");
+    { 
+      if (uClass.hasParentInterface() == false){
```

Comment:
@amindzak Please check the issue #840.
You may be able to fix it easily in this part of the code.